src/configure.py

```diff
@@ -766,13 +755,50 @@ def load_config( config ):
             if gpus[temp[0]] != "": LOGGER.warning("The original value will be overwritten. <%s>: %s --> %s"%(temp[0], gpus[temp[0]], temp[1]))
             gpus[temp[0]] = temp[1]
         else:
-            try:
+            if len(temp) >= 2:
                paths[temp[0]] = temp[1]
-            except:
-               paths[temp[0]] = ''
+            else:                               # key without value
+               paths[temp[0]] = ""
 
     return paths, compilers, flags, gpus
 
+def load_setting():
+    """
+    Return the default machine setting in the following order:
+    1. Read the local  setting located at `GAMER_LOCAL_SETTING`.
+    2. Read the global setting located at `GAMER_GLOBAL_SETTING`.
+    3. Fall back to `eureka_intel` for backward compatibility.
```

Comment:
IIUC, this routine is general-purpose and can load settings beyond just `machine`. If so, consider revising the description to make it more general.